configure.ac

```diff
@@ -263,6 +263,9 @@ AS_IF([test "$enable_debug" != ""], [
   AC_MSG_NOTICE([Enabling all debug options])
   enable_checks="3"
   OPT_FLAGS="-g -O0 -fno-inline -hipa1"
+  # Only for not gcc add -Og due to
+  # https://github.com/boutproject/BOUT-dev/issues/1879
+  $MPICXX --version | grep gcc -q || OPT_FLAGS="$OPT_FLAGS -Og"
```

Comment:
Note that this looks for `gcc`, but should be looking for `g++`. This seems pretty robust across all the systems and variations of GCC I've got access to at least. The current form fails to detect GCC across any of them.